tests/datasets/validators/test_dataset_validator.py

```diff
@@ -0,0 +1,258 @@
+            @property
+            def outputs(self):
+                return {"y"}
+        
+        validator = ConcreteValidator()
+        dataset = MockDataset(path=str(tmp_path), organism=Organism.HUMAN)
+        
+        try:
+            validator.validate_dataset(dataset)
+        except ValueError:
+            pytest.fail("validate_dataset raised ValueError unexpectedly!")
+
+# Test Cases for BaseSingleCellValidator
+class TestBaseSingleCellValidator:
+
+    @pytest.fixture
+    def sc_dataset(self, tmp_path) -> SingleCellDataset:
+        """Fixture to create a valid SingleCellDataset instance for testing."""
+        # Create a dummy dataset instance.
+        # We manually assign adata to avoid file I/O.
+        dataset = SingleCellDataset(path=str(tmp_path), organism=Organism.HUMAN)
+        dataset.adata = ad.AnnData(
+            X=np.random.poisson(size=(10, 10)),
+            obs=pd.DataFrame(index=[f"cell_{i}" for i in range(10)]),
+            var=pd.DataFrame(index=[f"{Organism.HUMAN.prefix}_gene_{i}" for i in range(10)]),
+        )
+        # Monkeypatch the 'inputs' property, which is required by the validator.
+        # It must be a dict to work with the `.keys()` call in the source.
+        dataset.inputs = {"X": True}
+        return dataset
+
+    def test_successful_instantiation_sc(self):
+        """Tests successful instantiation of a BaseSingleCellValidator subclass."""
+        class ConcreteSCValidator(BaseSingleCellValidator):
+            available_organisms: ClassVar[List[Organism]] = [Organism.HUMAN]
+            required_obs_keys: ClassVar[List[str]] = []
+            required_var_keys: ClassVar[List[str]] = []
+            @property
+            def inputs(self):
+                return {"X"}
+            @property
+            def outputs(self):
+                return set()
+        
+        validator = ConcreteSCValidator()
+        assert isinstance(validator, BaseSingleCellValidator)
+
+    def test_missing_available_organisms(self):
+        """Tests TypeError for missing available_organisms."""
+        with pytest.raises(TypeError, match="without available_organisms class variable"):
+            class InvalidSCValidator(BaseSingleCellValidator):
+                required_obs_keys: ClassVar[List[str]] = []
+                required_var_keys: ClassVar[List[str]] = []
+                @property
+                def inputs(self): return set()
+                @property
+                def outputs(self): return set()
+
+    def test_missing_required_obs_keys(self):
+        """Tests TypeError for missing required_obs_keys."""
+        with pytest.raises(TypeError, match="without required_obs_keys class variable"):
+            class InvalidSCValidator(BaseSingleCellValidator):
+                available_organisms: ClassVar[List[Organism]] = [Organism.HUMAN]
+                required_var_keys: ClassVar[List[str]] = []
+                @property
+                def inputs(self): return set()
```

Comment:
If you really want to be compact, this is `inputs = property(lambda self: set())`. Otherwise can you format this in the usual way?